src/virtualenv/app_data/via_disk_folder.py

```diff
@@ -155,6 +157,14 @@ def write(self, content):
             folder.mkdir(parents=True, exist_ok=True)
         except OSError:
             pass
+        with tempfile.NamedTemporaryFile("w", prefix="atomic-write", delete=False) as tf:
+            tf.write(ensure_text(json.dumps(content, sort_keys=True, indent=2)))
+        try:
```

Comment:
do we need this remove, wouldn't the rename, overwrite? If the rename fails we just deleted the existing file, making this write not that atomic...

Reply:
Windows prevents the file to be renamed when the target file exists. 😞

Reply:
Should this than be windows only logic, rather than universal? 🤔

Reply:
Yes, it's Windows only.